File: Core/Entities/Material.cs
namespace Core.Entities
{
    public class Material : BaseEntity
    {
        public string Name { get; set; }
        public MaterialType MaterialType { get; set; }
        public int MaterialTypeId { get; set; }
        public int Thickness { get; set; }
        public Color Color { get; set; }
        public int ColorId { get; set; }
        public decimal PricePerMeter { get; set; }
    }
}

File: client/src/app/shared/models/material.ts
export interface IMaterial {
    id: number;
    name: string;
    materialType: string;
    materialTypeId: number;
    thickness: number;
    mainColor: string;
    color: string;
    pictureUrl: string;
    colorId: number;
    pricePerMeter: number;
}

export interface IMaterialToCreate {
    name: string;
    materialTypeId: number;
    thickness: number;
    colorId: number;
    pricePerMeter: number;
}

export class MaterialFormValues implements IMaterialToCreate {
    name: string;
    materialTypeId: number;
    thickness = 0;
    colorId: number;
    pricePerMeter = 0;

    constructor(init?: MaterialFormValues) {
        Object.assign(this, init);
    }
}